src/components/UserContext.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { createContext, useState, useEffect } from "react";
+
+export const UserContext = createContext();
+
+export const UserProvider = ({ children }) => {
+  const [loggedInUser, setLoggedInUser] = useState(() => {
+    const savedUser = localStorage.getItem("user");
```

Comment:
While this is a way of stopping a user being logged out on a refresh, please don't access a user's local storage without informing them that it is happening. Especially without a way to remove it. Now if I clear my browser cache it should go. I however had no idea you've just accessed my local storage before I got to this part of your repo.